pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1260,6 +1266,23 @@ func generateDeprovisionRequest(cd *hivev1.ClusterDeployment) *hivev1.ClusterDep
 	return req
 }
 
+func generatePullSecretObj(pullSecret string, pullSecretName string, cd *hivev1.ClusterDeployment) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-pull-secret", pullSecretName),
```

Comment:
The full pull secret name has already been supplied. The suffix should not be added here.

Reply:
good catch.